drift/lib/src/runtime/executor/helpers/engines.dart

```diff
@@ -188,8 +188,10 @@ class _StatementBasedTransactionExecutor extends _TransactionExecutor {
       _StatementBasedTransactionExecutor this._parent, int depth)
       : _delegate = _parent._delegate,
         _startCommand = 'SAVEPOINT s$depth',
-        _commitCommand = 'RELEASE s$depth',
-        _rollbackCommand = 'ROLLBACK TO s$depth',
+        _commitCommand = _parent._db.dialect == SqlDialect.mariadb
+            ? 'RELEASE SAVEPOINT s$depth' : 'RELEASE s$depth',
+        _rollbackCommand = _parent._db.dialect == SqlDialect.mariadb
+            ? 'ROLLBACK TO SAVEPOINT s$depth' : 'ROLLBACK TO s$depth',
```

Comment:
I think we should be moving these to the `NoTransactionDelegate` class which is created by the connection implementation. The class could have a `String Function(int depth)` field for savepoints, releases and rollbacks. Here, we could then call `_startCommand = _parent._delegate.savepoint(depth)`, `_commitCommand = _parent._delegate.release(depth)` and so on.

Reply:
Good idea, check my implementation on my last commit if you agree with it and its documentation.